smarts/core/vehicle.py

```diff
@@ -394,9 +394,14 @@ def build_agent_vehicle(
             SceneColors.Agent.value if trainable else SceneColors.SocialAgent.value
         )
 
-        if agent_interface.vehicle_type == "sedan":
+        if mission.vehicle_spec == None:
+            vehicle_type = "sedan"
+        else:
+            vehicle_type = mission.vehicle_spec.veh_config_type
```

Comment:
is this the right place to get the vehicle type?

Reply:
Yes, although I am slightly confused as to why the `vehicle_spec` is inside the mission rather than in the plan.

Reply:
where should `vehicle_spec` be specified then?